universal-application-tool-0.0.1/app/controllers/LoginController.java

```diff
@@ -33,68 +30,65 @@
 
   private final HttpActionAdapter httpActionAdapter;
 
   private final Config config;
 
-  private final String applicantIdp;
-
   @Inject
   public LoginController(
-      @AdOidcClient @Nullable OidcClient adClient,
-      @IdcsOidcClient @Nullable OidcClient idcsClient,
-      @LoginRadiusSamlClient @Nullable SAML2Client loginRadiusClient,
+      @AdminAuthClient @Nullable IndirectClient adminClient,
+      @ApplicantAuthClient @Nullable IndirectClient applicantClient,
       SessionStore sessionStore,
       Config config) {
-    this.idcsClient = idcsClient;
-    this.adClient = adClient;
-    this.loginRadiusClient = loginRadiusClient;
+    this.adminClient = adminClient;
+    this.applicantClient = applicantClient;
     this.sessionStore = Preconditions.checkNotNull(sessionStore);
-    this.applicantIdp = checkNotNull(config).getString("auth.applicant_idp");
     this.httpActionAdapter = PlayHttpActionAdapter.INSTANCE;
     this.config = config;
   }
 
-  public Result loginWithRedirect(Http.Request request, Optional<String> redirectTo) {
-    if (applicantIdp.equals(AuthIdentityProviderName.LOGIN_RADIUS_APPLICANT.getString())) {
-      return loginRadiusLoginWithRedirect(request, redirectTo);
-    }
-    return idcsLoginWithRedirect(request, redirectTo);
-  }
-
-  public Result idcsLogin(Http.Request request) {
-    return login(request, idcsClient);
+  public Result adminLogin(Http.Request request) {
+    return login(request, adminClient);
   }
 
-  public Result idcsLoginWithRedirect(Http.Request request, Optional<String> redirectTo) {
+  public Result applicantLogin(Http.Request request, Optional<String> redirectTo) {
     if (redirectTo.isEmpty()) {
-      return idcsLogin(request);
+      return login(request, applicantClient);
     }
-    return login(request, idcsClient)
+    return login(request, applicantClient)
         .addingToSession(request, REDIRECT_TO_SESSION_KEY, redirectTo.get());
   }
 
-  public Result loginRadiusLogin(Http.Request request) {
-    return login(request, loginRadiusClient);
-  }
+  public Result register(Http.Request request) {
+    String idp;
+    try {
+      idp = config.getString("auth.applicant_idp");
+    } catch (ConfigException.Missing e) {
+      // Default to IDCS.
+      idp = AuthIdentityProviderName.IDCS_APPLICANT.toString();
+    }
 
-  public Result loginRadiusLoginWithRedirect(Http.Request request, Optional<String> redirectTo) {
-    if (redirectTo.isEmpty()) {
-      return loginRadiusLogin(request);
+    boolean isIDCS = idp.equals(AuthIdentityProviderName.IDCS_APPLICANT.toString());
+
+    // Because this is only being called when we know IDCS is available, this route should
+    // technically
+    // never happen.
+    if (!isIDCS) {
+      return login(request, applicantClient);
     }
-    return login(request, loginRadiusClient)
-        .addingToSession(request, REDIRECT_TO_SESSION_KEY, redirectTo.get());
+
+    return idcsRegister(request);
```

Comment:
@shanemc-goog do these changes look like what you had in mind?

Reply:
Per the team discussion I believe you'll have the existing url routes internally use your new controller.

Reply:
Should be fixed, let me know what you think